tsl/src/bgw_policy/job.c

```diff
@@ -150,34 +153,85 @@ execute_drop_chunks_policy(int32 job_id)
 {
 	bool started = false;
 	BgwPolicyDropChunks *args;
+	int64 older_than_datum;
+	Oid table_relid;
+	Oid now_func;
+	Oid rettype;
+	Hypertable *hypertable;
+	Cache *hcache;
+	Dimension *open_dim;
 
 	if (!IsTransactionOrTransactionBlock())
 	{
 		started = true;
 		StartTransactionCommand();
+		PushActiveSnapshot(GetTransactionSnapshot());
 	}
 
 	/* Get the arguments from the drop_chunks_policy table */
 	args = ts_bgw_policy_drop_chunks_find_by_job(job_id);
-
+	table_relid = ts_hypertable_id_to_relid(args->fd.hypertable_id);
 	if (args == NULL)
 		ereport(ERROR,
 				(errcode(ERRCODE_TS_INTERNAL_ERROR),
 				 errmsg("could not run drop_chunks policy #%d because no args in policy table",
 						job_id)));
 
-	ts_chunk_do_drop_chunks(ts_hypertable_id_to_relid(args->fd.hypertable_id),
-							IntervalPGetDatum(&args->fd.older_than),
-							0,
-							INTERVALOID,
-							InvalidOid,
-							args->fd.cascade,
-							args->fd.cascade_to_materializations,
-							LOG);
+	if (args->fd.older_than.is_time_interval)
+	{
+		ts_chunk_do_drop_chunks(table_relid,
+								IntervalPGetDatum(&args->fd.older_than.time_interval),
+								(Datum) 0,
+								INTERVALOID,
+								InvalidOid,
+								args->fd.cascade,
+								args->fd.cascade_to_materializations,
+								LOG);
+	}
+	else
+	{
+		hcache = ts_hypertable_cache_pin();
+		hypertable = ts_hypertable_cache_get_entry(hcache, table_relid);
+		/* First verify that the hypertable corresponds to a valid table */
+		if (hypertable == NULL)
+			ereport(ERROR,
+					(errcode(ERRCODE_TS_HYPERTABLE_NOT_EXIST),
+					 errmsg("could not set integer_now function because \"%s\" is not a hypertable",
+							get_rel_name(table_relid))));
+
+		open_dim = hyperspace_get_open_dimension(hypertable->space, 0);
+		rettype = ts_dimension_get_partition_type(open_dim);
+
+		Assert(IS_INTEGER_TYPE(rettype));
+
+		now_func = get_integer_now_func(table_relid);
```

Comment:
should take dimension and return now function

Reply:
refactor all to ___(open_Dim Dimension ,invl ts_interval ) : now - older_than_datum